client/my-sites/plans/current-plan/purchases-listing.jsx

```diff
@@ -124,7 +124,52 @@ class PurchasesListing extends Component {
 		return null;
 	}
 
-	getExpirationInfo( purchase ) {
+	getPurchaseByCurrentPlan() {
+		const { currentPlanSlug, purchases } = this.props;
+
+		if ( ! currentPlanSlug ) {
+			return null;
+		}
+
+		return find(
+			purchases,
+			purchase => purchase.active && purchase.productSlug === currentPlanSlug
+		);
+	}
+
+	getExpirationInfoForPlan( plan ) {
+		// No expiration date for free plan or partner site.
+		if ( this.isFreePlan( plan ) || isPartnerPurchase( plan ) ) {
+			return null;
+		}
+
+		const subscribedMoment = plan.subscribedDate ? this.props.moment( plan.subscribedDate ) : null;
+
+		const expiryMoment = plan.expiryDate ? this.props.moment( plan.expiryDate ) : null;
+
+		const renewMoment =
+			plan.autoRenew && plan.autoRenewDate ? this.props.moment( plan.autoRenewDate ) : null;
+
+		let isRefundable = false;
+
+		// we need to find the purchase linked to the current plan to find out if it's refundable or not
+		const purchaseForCurrentPlan = this.getPurchaseByCurrentPlan();
```

Comment:
Technically you're fixing a bug here, but in practice I think it makes the UI worse.

If you have either a WordPress.com plan or a Jetpack plan, then before this diff you'll see "Renews on [Date]" when you go to `plans/my-plan/[site]`.  After this diff, you'll see "Purchased on [Date]" there.  I don't think that's a good change.  The user is already on a page that is designed to show their current purchases, so they don't need extra confirmation specifically telling them that they purchased it.

I think we should actually go the other direction and stop showing "Purchased on [Date]" for Jetpack Backup on this page too, to make it consistent with the others.

I tried to figure out how this change got made in the first place and I think I see what happened:
1. In the threaded discussion underneath p8oabR-qW-p2#comment-3622 there was a discussion of conditionally showing "Purchased on [Date]" for Jetpack Backup on the *actual* plans page (`plans/[site]`), and it probably does make some sense there.  (That is the page you go to in order to buy new things, so if you just bought Jetpack Backup recently it is helpful to see that specifically.)
2. Somehow when it was implemented, though, it got added to both `plans/[site]` and `plans/my-plan/[site]`.  I think that was a mistake.

Bottom line: My suggestion is that for the two usages of the `<ProductExpiration>` component in this file, we shouldn't pass in either `purchaseDateMoment` or `isRefundable` at all.  That would have the desired effect of showing only the "Renews"/"Expires" text at  `plans/my-plan/[site]`, while still conditionally showing the "Purchased" text at `plans/[site]`.

If that makes sense, I'd be happy to leave a comment on the above P2 thread explaining my reasoning and with a link back here.

Reply:
Thanks. Implemented.